rest-api/dao/participant_dao.py

```diff
@@ -114,25 +114,43 @@ def _do_update(self, session, obj, existing_obj):
                             else None)
       need_new_summary = True
 
-    if obj.organizationId or obj.siteId or obj.hpoId:
-      site, organization, awardee = self.get_pairing_level(obj)
-      obj.organizationId = organization
-      obj.siteId = site
-      obj.hpoId = awardee
-      if awardee:
-        # get provider link for hpo_id (awardee)
-        obj.providerLink = make_primary_provider_link_for_id(awardee)
+    update_pairing = True
+    if obj.siteId is None and obj.organizationId is None and obj.hpoId is None and \
+      obj.providerLink == 'null':
+      update_pairing = False
+
+    if update_pairing == True:
+      if obj.organizationId or obj.siteId or obj.hpoId:
+        site, organization, awardee = self.get_pairing_level(obj)
+        obj.organizationId = organization
+        obj.siteId = site
+        obj.hpoId = awardee
+        if awardee is not None and (obj.hpoId != existing_obj.hpoId):
```

Comment:
So the logic if I understand it is: if awardee changes, have that take priority. If it doesn't, but providerLink changes, have that take priority. Is that right?

We should add comments to that effect in here. It's kind of weird behavior. (Given that we patch existing participants, I can see why you'd do it that way, but it needs documentation.)